src-java/kilda-persistence-neo4j/src/main/java/org/openkilda/persistence/repositories/impl/Neo4JMirrorGroupRepository.java

```diff
@@ -0,0 +1,89 @@
+/* Copyright 2019 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.persistence.repositories.impl;
+
+import static java.lang.String.format;
+
+import org.openkilda.model.GroupId;
+import org.openkilda.model.MirrorGroup;
+import org.openkilda.model.PathId;
+import org.openkilda.model.SwitchId;
+import org.openkilda.persistence.PersistenceException;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.MirrorGroupRepository;
+
+import com.google.common.collect.ImmutableMap;
+import org.neo4j.ogm.cypher.ComparisonOperator;
+import org.neo4j.ogm.cypher.Filter;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+
+public class Neo4JMirrorGroupRepository extends Neo4jGenericRepository<MirrorGroup> implements MirrorGroupRepository {
+    static final String PATH_ID_PROPERTY_NAME = "path_id";
+
+    public Neo4JMirrorGroupRepository(Neo4jSessionFactory sessionFactory, TransactionManager transactionManager) {
+        super(sessionFactory, transactionManager);
+    }
+
+    @Override
+    public Collection<MirrorGroup> findByPathId(PathId pathId) {
+        Filter pathIdFilter = new Filter(PATH_ID_PROPERTY_NAME, ComparisonOperator.EQUALS, pathId);
+
+        Collection<MirrorGroup> groups = loadAll(pathIdFilter);
+        if (groups.size() > 1) {
+            throw new PersistenceException(format("Found more that 2 Group entity by path (%s). "
```

Comment:
`more that 1 Group`

I think it will be helpful to add information about these extra groups. Something like 
`"Found more that 1 Group entity by path (%s). Group IDs: [%s]"`